File: test/language/default_values/java/default_values/StructureDefaultValuesTest.java
package default_values;

import static org.junit.Assert.*;

import org.junit.Test;

import default_values.structure_default_values.StructureDefaultValues;
import default_values.structure_default_values.BasicColor;
import default_values.structure_default_values.Permission;

public class StructureDefaultValuesTest
{
    @Test
    public void checkDefaultBoolValue()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals(true, structureDefaultValues.getBoolValue());
    }

    @Test
    public void checkDefaultBit4Value()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals(0x0F, structureDefaultValues.getBit4Value().byteValue());
    }

    @Test
    public void checkDefaultInt16Value()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals(0x0BEE, structureDefaultValues.getInt16Value());
    }

    @Test
    public void checkDefaultFloat16Value()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals(1.23f, structureDefaultValues.getFloat16Value(), 0.00001f);
    }

    @Test
    public void checkDefaultFloat32Value()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals(1.234f, structureDefaultValues.getFloat32Value(), 0.00001f);
    }

    @Test
    public void checkDefaultFloat64Value()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals(1.2345, structureDefaultValues.getFloat64Value(), 0.000000001);
    }

    @Test
    public void checkDefaultStringValue()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals("string", structureDefaultValues.getStringValue());
    }

    @Test
    public void checkDefaultEnumValue()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals(BasicColor.BLACK, structureDefaultValues.getEnumValue());
    }

    @Test
    public void checkDefaultBitmaskValue()
    {
        final StructureDefaultValues structureDefaultValues = new StructureDefaultValues();
        assertEquals(Permission.Values.READ_WRITE, structureDefaultValues.getBitmaskValue());
    }
}

File: test/language/default_values/cpp/StructureDefaultValuesTest.cpp
#include "gtest/gtest.h"

#include "zserio/BitStreamWriter.h"
#include "zserio/BitStreamReader.h"
#include "zserio/CppRuntimeException.h"

#include "default_values/structure_default_values/StructureDefaultValues.h"
#include "default_values/structure_default_values/BasicColor.h"

namespace default_values
{
namespace structure_default_values
{

TEST(StructureDefaultValuesTest, checkDefaultBoolValue)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(true, structureDefaultValues.getBoolValue());
}

TEST(StructureDefaultValuesTest, checkDefaultBit4Value)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(0x0F, structureDefaultValues.getBit4Value());
}

TEST(StructureDefaultValuesTest, checkDefaultInt16Value)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(0x0BEE, structureDefaultValues.getInt16Value());
}

TEST(StructureDefaultValuesTest, checkDefaultFloat16Value)
{
    StructureDefaultValues structureDefaultValues;
    float diff = 1.23f - structureDefaultValues.getFloat16Value();
    if (diff < 0.0f)
        diff = -diff;
    ASSERT_TRUE(diff <= std::numeric_limits<float>::epsilon());
}

TEST(StructureDefaultValuesTest, checkDefaultFloat32Value)
{
    StructureDefaultValues structureDefaultValues;
    float diff = 1.234f - structureDefaultValues.getFloat32Value();
    if (diff < 0.0f)
        diff = -diff;
    ASSERT_TRUE(diff <= std::numeric_limits<float>::epsilon());
}

TEST(StructureDefaultValuesTest, checkDefaultFloat64Value)
{
    StructureDefaultValues structureDefaultValues;
    float diff = 1.2345 - structureDefaultValues.getFloat64Value();
    if (diff < 0.0)
        diff = -diff;
    ASSERT_TRUE(diff <= std::numeric_limits<double>::epsilon());
}

TEST(StructureDefaultValuesTest, checkDefaultStringValue)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ("string", structureDefaultValues.getStringValue());
}

TEST(StructureDefaultValuesTest, checkDefaultEnumValue)
{
    StructureDefaultValues structureDefaultValues;
    ASSERT_EQ(BasicColor::BLACK, structureDefaultValues.getEnumValue());
}

} // namespace structure_default_values
} // namespace default_values
